### PR TITLE
Add List filter configuration

### LINKED ISSUE #633: Missing filter for article types
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes / port from feature of sulu 1.6 version
| SuluArticleBundle Version | 2.4
| Sulu Version | 2.5

#### Actual Behavior

The filters for article types are missing in the sulu 2.x version of the article bundle.

#### Expected Behavior

When choosing articles as teaser source, i want to be able to filter for the specific article types.

#### Possible Solutions

implement filters like in the old version: 

Bildschirmfoto 2022-10-27 um 09 02 36

#### TODO

- [x] Basic Filters added by #632 
- [ ] Parent Page ID

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #633
| Related issues/PRs | none
| License | MIT

#### What's in this PR?

This PR add the filter configuration for article-lists.

### CODE REVIEW

File: Metadata/ListMetadataVisitor.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ */
+
+namespace Sulu\Bundle\ArticleBundle\Metadata;
+
+use Sulu\Bundle\AdminBundle\Metadata\ListMetadata\ListMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\ListMetadata\ListMetadataVisitorInterface;
+use Sulu\Component\Content\Compat\Structure\StructureBridge;
+use Sulu\Component\Content\Compat\StructureManagerInterface;
+
+class ListMetadataVisitor implements ListMetadataVisitorInterface
```
Comment (alexander-schranz) [inline Metadata/ListMetadataVisitor.php:24]: mark services `@internal`

```suggestion
/**
 * @final
 * @internal
 */
class ListMetadataVisitor implements ListMetadataVisitorInterface
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

File: Controller/ArticleController.php [hunk 2]
```diff
@@ -212,6 +212,20 @@ public function cgetAction(Request $request): Response
...
+         *         to: string,
+         *     },
+         *     type?: string,
+         *     contactId?: string,
+         *     categoryId?: string,
+         *     tagId?: string,
+         *     pageId?: string,
+         *     publishedState?: string,
+         *  } $filter */
+        $filter = $request->get('filter');
```
Comment (alexander-schranz) [inline Controller/ArticleController.php]:
```suggestion
        $filter = $request->query->get('filter');
```

Comment (wachterjohannes) [reply Controller/ArticleController.php]: `$request->query->get('filter')` cannot return an array 🙈

Comment (alexander-schranz) [reply Controller/ArticleController.php]:
```php
$request->query->all('filter')
```

Comment (alexander-schranz) [reply Controller/ArticleController.php]: looks like does not exist on 4.4 so:

```php
$request->query->all()['filter'] ?? []
```

that is similar to the request get

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

**Comment by alexander-schranz** [comment]
Should we target this as a new feature (2.5 not released yet) or as a bugfix (2.4)? /cc @chirimoya

**Comment by wachterjohannes** [comment]
@alexander-schranz is a bugfix for me 😂

**Comment by chirimoya** [comment]
@wachterjohannes @alexander-schranz  `2.4`

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**